package.json

```diff
@@ -159,13 +159,14 @@
     "html-loader": "^1.1.0",
     "html-webpack-plugin": "^4.0.0-beta.2",
     "ignore-loader": "^0.1.2",
+    "immer": "^9.0.7",
```

Comment:
can you tell me a bit about `immer`?

Reply:
Immer is a library that makes it a bit easier to work with immutable state (in this case, the toggle trees). Basically it eliminates the use of spread operators to create a copy of an object, then modify it, then reset the state object to the copy. Now, since we are cashing a stringified version of the toggle tree in this case, immer isn't strictly necessary. We won't run into issues writing mutable code since its just parsed and stringified on every update. I just included for good practice and in case we ever change how we store the toggle trees in the future (its just as efficient). Can remove it easily if you want.